vault/identity_store_util.go

```diff
@@ -137,15 +143,12 @@ func (i *IdentityStore) loadGroups(ctx context.Context) error {
 			nsCtx := namespace.ContextWithNamespace(ctx, ns)
 
 			// Ensure that there are no groups with duplicate names
-			groupByName, err := i.MemDBGroupByName(nsCtx, group.Name, false)
+			groupByName, err := i.MemDBGroupByName(nsCtx, group.Name, true)
```

Comment:
I think the clone here is correct even without renames, right? Do we want to do this in a separate PR with some docs about why?